main.rb

```diff
@@ -0,0 +1,122 @@
+# Enumerable
+module Enumerable
+  # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+
+  def my_each
+    return to_enum(__method__) unless block_given?
+
+    if is_a?(Array)
+      size.times do |i|
+        yield self[i]
+      end
+    else size.times do |i|
+      yield [keys[i], values[i]]
+    end
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(__method__) unless block_given?
+
+    size.times do |i|
+      yield self[i], i
+    end
+  end
+
+  def my_select
+    return to_enum(__method__) unless block_given?
+
+    array = []
+    my_each do |i|
+      array << i if yield i
+    end
+    array
+  end
+
+  def my_all?(*args)
+    if block_given?
+      my_each { |i| return false unless yield i }
+    elsif args != [] && args[0].class != Regexp
+      my_each { |i| return false unless i.is_a?(args[0]) }
+    elsif args[0].class == Regexp
+      my_each { |i| return false if (args[0] =~ i).nil? }
+    else
+      my_each { |i| return false unless i.nil? }
+    end
+    true
+  end
+
+  def my_any?(*args)
+    if block_given?
+      my_each { |i| return true if yield i }
+    elsif args != [] && args[0].class != Regexp
+      my_each { |i| return true if i.is_a?(args[0]) }
+    elsif args[0].class == Regexp
+      my_each { |i| return true unless (args[0] =~ i).nil? }
+    else
+      my_each { |i| return true if i.nil? }
+    end
+    false
+  end
```

Comment:
- [ ] should have working `my_any?` identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

 #### my_any? 

- [ ] when no block or argument is given returns true if at least one of the collection is not false or nil

```ruby     
[nil, false, false].my_any? == [nil, false, false].any? => false
```
- Yours displays the wrong answer
![image](https://user-images.githubusercontent.com/52209067/85830414-ca22df00-b749-11ea-9ad1-9b02943738b3.png)

- Real any?
![image](https://user-images.githubusercontent.com/52209067/85830468-e161cc80-b749-11ea-9293-87540d9797d9.png)

- [ ] when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern

```ruby
[3,4,7,11].my_any?(3) should be equal to this [3,4,7,11].any?(3) => true
```
- Yours retrieves an error
![image](https://user-images.githubusercontent.com/52209067/85830706-41587300-b74a-11ea-9845-13af49d1c605.png)

- Real any?
![image](https://user-images.githubusercontent.com/52209067/77689883-10f51300-6f68-11ea-8a4c-319977bb3842.png)